### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by katfogy**
**In this pull request I;**
- [ ] Write integration tests with Capybara gem.
- [ ] Secure app from n+1 problems.

Before

After

Before

After

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @katfogy,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] This is a peer-programming project milestone and I can see only one team member has committed all changes, So please make sure that all team members have the same amount of work shared and committed to GitHub.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @katfogy and @ClarenceAtim,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Modifying only the README does not count as a contribution to the project. You need to make actual commits to the code of the project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/integrations/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      posts_count: 5
+    )
+  end
+
+  it 'redirect to the user\'s show page' do
+    visit users_path
+    click_link(href: user_path(@user1.id))
+    expect(page).to have_current_path(user_path(@user1.id))
+  end
+end
```
Comment (TedLivist) [inline spec/integrations/user_index_spec.rb]: - [ ] Kindly look at the project guideline. Each test for the user index page should have its own `it-block`. You have added only one test, kindly add the remaining three.

File: spec/integrations/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    visit user_path(@user1.id)
+    expect(page).to have_button('See All Posts')
+  end
+
+  it 'redirect to the post\'s show page' do
+    visit user_path(@user1.id)
+    click_link(href: user_post_path(@user1.id, 4))
+    expect(page).to have_current_path(user_post_path(@user1.id, 4))
+  end
+end
```
Comment (TedLivist) [inline spec/integrations/user_show_spec.rb]: - [ ] Kindly add the remaining tests for the user show page.

  - Tests that asserts you can see the user's profile picture.
  - Tests that asserts you can see the number of posts the user has written.
  - Tests that asserts you can see the user's first 3 posts.
  - Tests that asserts you can see a button that lets me view all of a user's posts.
  - Tests that redirects to the user's post's index page when you click on a see all posts.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @katfogy and @ClarenceAtim,

Good job so far!
✔️ Well-written and passing integration tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @katfogy , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Added Integration Test ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**